addons/product/views/product_supplierinfo_views.xml

```diff
@@ -91,7 +91,7 @@
         <field name="arch" type="xml">
             <list string="Vendor Information" multi_edit="1">
                 <field name="sequence" widget="handle"/>
-                <field name="partner_id" readonly="1"/>
+                <field name="partner_id" readonly="1" widget="many2one_avatar_user"/>
```

Comment:
Isn't it the `res_partner_many2one` that you want here? :thinking: 

This field does not hold a `res.users` but a `res.partner`